libcperciva/network/network_connect.c

```diff
@@ -178,18 +182,26 @@ network_connect(struct sock_addr * const * sas,
     int (* callback)(void *, int), void * cookie)
 {
 
-	/* Let network_connect_timeo handle this. */
-	return (network_connect_timeo(sas, NULL, callback, cookie));
+	/* Let network_connect_internal handle this. */
+	return (network_connect_internal(sas, NULL, NULL, callback, cookie));
 }
 
 /**
- * network_connect_timeo(sas, timeo, callback, cookie):
- * Behave as network_connect(), but wait a duration of at most ${timeo} for
- * each address which is being attempted.
+ * network_connect_internal(sas, sa_b, timeo, callback, cookie):
```

Comment:
If we move this function above `network_connect`, we won't need the forward declaration.  Slightly more churn in the file but it ends up being cleaner.

Reply:
Oh, I was aiming for minimal diffs.  Ok, I'll move it.